### PR TITLE
Backport of PR #4698 to the 1.0.2

### PR DESCRIPTION
**Pull request by PavelKopyl**
##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: apps/ca.c [hunk 1]
```diff
@@ -1817,8 +1817,6 @@ static int do_body(X509 **xret, EVP_PKEY *pkey, X509 *x509,
 
             if (push != NULL) {
                 if (!X509_NAME_add_entry(subject, push, -1, 0)) {
-                    if (push != NULL)
-                        X509_NAME_ENTRY_free(push);
```
Comment (levitte) [inline apps/ca.c]: You still need to free it.  It's not freed down by `err:`

Comment (PavelKopyl) [reply apps/ca.c]: As far as I understand, it's freed among other stuff in X509_REQ_free(rreq) at apps/ca.c:1559. It's the same behaviour as on master.

Comment (mattcaswell) [reply apps/ca.c]: Yes, this is the same as per #4698

**Comment by mattcaswell** [comment]
Pushed. As with the master/1.1.0 version I edited the commit message to remove "CLA: trivial". A CLA is on file now anyway, so it is irrelevant.

**Pull request closed without merging**